docs/spec/spec.md

```diff
@@ -116,18 +116,28 @@ The conforming implementation of the specification is released and included in t
 11. [Security](#11-security)
     * 11.1 [Authentication and Authorization](#111-authentication-and-authorization)
         * 11.1.1 [Declarative Approach](#1111-declarative-approach)
-            * 11.1.1.1 [Basic Authentication - File User Store](#11111-basic-authentication---file-user-store)
-            * 11.1.1.2 [Basic Authentication - LDAP User Store](#11112-basic-authentication---ldap-user-store)
-            * 11.1.1.3 [JWT Authentication](#11113-jwt-authentication)
-            * 11.1.1.4 [OAuth2](#11114-oauth2)
+            * 11.1.1.1 [Listener - Basic Authentication - File User Store](#11111-listener---basic-authentication---file-user-store)
+            * 11.1.1.2 [Listener - Basic Authentication - LDAP User Store](#11112-listener---basic-authentication---ldap-user-store)
+            * 11.1.1.3 [Listener - JWT Authentication](#11113-listener---jwt-authentication)
+            * 11.1.1.4 [Listener - OAuth2](#11114-listener---oauth2)
+            * 11.1.1.5. [Client - Basic Authentication](#11115-client---basic-authentication)
+            * 11.1.1.6. [Client - Bearer Token Authentication](#11116-client---bearer-token-authentication)
+            * 11.1.1.7. [Client - Self Signed JWT Authentication](#11117-client---self-signed-jwt-authentication)
+            * 11.1.1.8. [Client - OAuth2](#11118-client---oauth2)
```

Comment:
On second thoughts, I think it is better to break this into two parts.
```
11.1.1 Declarative Approach
    * 11.1.1.1 Service
        * 11.1.1.1.1 Basic Authentication - File User Store
        * ...
    *  11.1.1.2 Client
        * 11.1.1.2.1. Basic Authentication
        * ...
```

Looks cleaner for me. WDYT?

Reply:
+1. will do the same to SSL stuff

Reply:
Since we don't have imperative approach for client followed the below structure
```Security
11.1  Service Authentication and Authorization
    11.1.1 Declarative
    11.1.2 Imperative
11.2 Client Authentication and Authorization
11.3 SSL/TLS and Mutual SSL
   11.3.1 Listener
   11.3.2 Client
```